### PR TITLE
Handle that ContentType might be null

### LINKED ISSUE #136: ModernHttpClient crashes on Android when used in SignalR
Using ModernHttpClient and SignalR together gives great benefits related to performance etc and also having the connection reconnecting when switching from wifi to 3g/4g.

To use ModernHttpClient together with SignalR my solution was to subclass the DefaultHttpClient from SignalR and use that instance as input to the Start method.
```C#
public class ModernHttpClientSignalR : DefaultHttpClient
{
        protected override HttpMessageHandler CreateHandler()
        {
            return new NativeMessageHandler();
        }
}

await _connection.Start(new ModernHttpClientSignalR());
```
This works nicely on iOS.
 On Android the ModernHttpClient does not like it when SignalR is posting data without any ContentType and I had to fix line 83 in OkHttpNetworkHandler.cs.

From: 
```C#
var contentType = String.Join (" ", request.Content.Headers.GetValues ("Content-Type"));
```
To:
```C#
var contentType = string.Empty;
if (request.Content.Headers.ContentType != null)
{
    contentType = String.Join(" ", request.Content.Headers.GetValues("Content-Type"));
}
```
It would be great if this fix could be part of the original code.

### PR DESCRIPTION
**Pull request by peterno**
When using ModernHttpClient with SignalR it crashes on Android because SignalR does not set content type when posting data to the server. 
If ContentType is null then string.Empty is used for MediaType.Parse and it thereby returns null and null is valid for RequestBody.Create.

Fixes #136

### CODE REVIEW

File: src/ModernHttpClient/Android/OkHttpNetworkHandler.cs [hunk 1]
```diff
@@ -80,7 +80,11 @@ private string GetHeaderSeparator(string name)
             if (request.Content != null) {
                 var bytes = await request.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
 
-                var contentType = String.Join (" ", request.Content.Headers.GetValues ("Content-Type"));
+                var contentType = string.Empty;
+                if (request.Content.Headers.ContentType != null)
+                {
```
Comment (paulcbetts) [inline src/ModernHttpClient/Android/OkHttpNetworkHandler.cs]: Move this bracket up to the previous line

**Comment by paulcbetts** [comment]
Great PR! I'll make these reverts go away, throw in an extra commit to fix the bracket

**Comment by peterno** [comment]
Bracket moved, thanks for your fast response. Amazing work Paul!

**Comment by paulcbetts** [comment]
Merged by hand - thanks @peterno!

**Comment by peterno** [comment]
Unfortunately it does not work when using it with SignalR and 2.4.0. I tried to change back the OkHttp.dll to the one in 2.3.0 but that gave other errors with the Builder.

My linkersettings is None and i am using Xamarin.Andriod: 4.20.2.1 and Xamarin 3.9.547.0 in Visual Studio.

I have to switch back to 2.3.0 and my own fork of the Android version.

Stacktrace is:
```
Missing method Android.Runtime.JNIEnv::CallObjectMethod(intptr,intptr,JValue*) in assembly Mono.Android.dll, referenced in assembly /storage/emulated/0/Android/data/Trio.Agent/files/.__override__/OkHttp.dll
Exception: System.MissingMethodException: Method not found: 'Android.Runtime.JNIEnv.CallObjectMethod'.
  at ModernHttpClient.NativeMessageHandler+<SendAsync>c__async0.MoveNext () [0x00154] in /Users/paul/code/paulcbetts/modernhttpclient/src/ModernHttpClient/Android/OkHttpNetworkHandler.cs:96 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[System.Net.Http.HttpResponseMessage].GetResult () [0x00000] in <filename unknown>:0 
  at System.Net.Http.HttpClient+<SendAsyncWorker>c__async0.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in <filename unknown>:0 
  
```

**Pull request closed without merging**